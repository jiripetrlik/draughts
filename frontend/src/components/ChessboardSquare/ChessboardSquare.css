table.chessboard td {
    width: 50px;
    height: 50px;
    border: 1px solid black;
}

table.chessboard td.light {
    background-color: beige;
}

table.chessboard td.dark {
    background-color: brown;
}
