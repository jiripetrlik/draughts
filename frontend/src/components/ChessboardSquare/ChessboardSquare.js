import React from "react"
import "./ChessboardSquare.css"
import Piece from "../Piece/Piece"


class ChessboardSquare extends React.Component {
    render() {
        return (
            <td className={this.color()}>
                {
                    this.props.pieceType !== "none" &&
                    <Piece type={this.props.pieceType}/>
                }
            </td>
        )
    }

    color() {
        let color

        if (this.props.row % 2 === 0) {
            if (this.props.column % 2 === 0) {
                color = "light"
            } else {
                color = "dark"
            }
        } else {
            if (this.props.column % 2 === 0) {
                color = "dark"
            } else {
                color = "light"
            }
        }

        return color
    }
}

export default ChessboardSquare;
