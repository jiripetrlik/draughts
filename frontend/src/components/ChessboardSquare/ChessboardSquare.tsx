import React from "react"
import "./ChessboardSquare.css"
import Piece from "../Piece/Piece"


class ChessboardSquare extends React.Component<Readonly<any>, Readonly<any>> {
    constructor(props: any) {
        super(props)
        this.handleClick = this.handleClick.bind(this)
    }

    color() {
        let color

        if (this.props.row % 2 === 0) {
            if (this.props.column % 2 === 0) {
                color = "light"
            } else {
                color = "dark"
            }
        } else {
            if (this.props.column % 2 === 0) {
                color = "dark"
            } else {
                color = "light"
            }
        }

        if (this.props.selected === "yes") {
            color = color + "selected"
        }

        return color
    }

    handleClick() {
        this.props.onHandleClick(this.props.row, this.props.column)
    }
    
    render() {
        return (
            <td className={this.color()} onClick={this.handleClick}>
                {
                    this.props.pieceType !== "none" &&
                    <Piece type={this.props.pieceType}/>
                }
            </td>
        )
    }
}

export default ChessboardSquare;
