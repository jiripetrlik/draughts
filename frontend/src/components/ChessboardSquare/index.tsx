import ChessboardSquare from "./ChessboardSquare"

export default ChessboardSquare
