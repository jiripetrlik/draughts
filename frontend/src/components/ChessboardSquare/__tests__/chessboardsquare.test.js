import React from 'react';
import renderer from 'react-test-renderer';
import ChessboardSquare from '../ChessboardSquare'

test('snapshot test', () => {
    const row = 1
    const column = 2
    const selected = "no"
    const pieceType = "white-piece"

    const tree = renderer
        .create(<ChessboardSquare row={row} column={column} selected={selected} pieceType={pieceType} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
