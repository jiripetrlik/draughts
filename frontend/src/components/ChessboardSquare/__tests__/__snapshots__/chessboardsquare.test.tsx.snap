// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
<td
  className="dark"
  onClick={[Function]}
>
  <img
    alt=""
    src="white-piece.svg"
  />
</td>
`;
